File: src/MinimalLSA/TermGeneration/ISourceTermGenerator.cs
using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

namespace MinimalLSA
{
	public interface ISourceTermGenerator
	{
		string[] GetTerms (string sourceTextLine);
		string[] GetTokens (string sourceTextLine);
		bool TryGetStems (string token, out string[] stems);
		List<string> StopWords { get; set; }
		Regex TokenPattern { get; set; }
		Regex WordMatchPattern { get; set; }
		string WordReplacePattern { get; set; }
	}
}

File: src/JLSA/jlsa/jlsa/ISourceTermGenerator.java
package jlsa;

import java.util.*;
import java.util.regex.*;

public interface ISourceTermGenerator 
{
	Pattern getTokenPattern();
	void setTokenPattern(Pattern value);
	
	Pattern getWordPattern();
	void setWordPattern(Pattern value);
	
	ArrayList<String> getStopWords();
	void setStopWords(ArrayList<String> value);
	
	boolean TryGetStems(String token, ArrayList<String> stems);
	String[] GetTerms(String sourceText);
	String[] GetTokens(String sourceText);
	
}
